implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -171,6 +171,19 @@ impl VaultsState {
         std::os::unix::fs::symlink(original, link)?;
         self.get(name)
     }
+
+    pub fn list(&self) -> Result<Vec<VaultState>> {
+        let mut vaults: Vec<VaultState> = vec![];
+        for entry in std::fs::read_dir(&self.dir)? {
+            let entry = entry?;
+            if entry.path().extension() == Some(OsStr::from_bytes(b"json")) {
```

Comment:
You don't need to check the extension, just try to load the vault with `self.get(vault_name)` and if that fails, then you skip it; if you can get it, then you add it to the `vaults` vec.

Reply:
Thank you for your input.

Switched to the already implemented `list()` function which contains the suggested changes. It was introduced with the commit adding the delete part.

https://github.com/build-trust/ockam/commit/7b1d00e8b11881d6b7c422f9e7acc4c77bbc7ba5#diff-a17a5a1da1448cd1d92017a87dd5cd59c347565441dea82b61b4903d814776d7R152